### ISSUE TITLE
reached unreachable code

### ISSUE THREAD

**Issue by r4gus** [david-vanderson/dvui#62]
I just encountered the following bug when using a Button with a Text, e.g.: https://github.com/r4gus/keypass/blob/4a480b8eda69ccd1be8549e1a823244996372283/src/gui/login.zig#L103

```
thread 5585 panic: reached unreachable code
/home/sugar/.cache/zig/p/1220bff2951dbed145107b567ad05ae4830bde54fb54d0d96f5297f7aeccfe7dc7c7/src/Options.zig:152:99: 0x322e16 in color (passkeez)
        .press_text => ret = cs.press_text orelse dvui.themeGet().style_content.press_text orelse unreachable,
                                                                                                  ^
/home/sugar/.cache/zig/p/1220bff2951dbed145107b567ad05ae4830bde54fb54d0d96f5297f7aeccfe7dc7c7/src/dvui.zig:7938:83: 0x35d5be in button (passkeez)
    if (captured(bw.wd.id)) options = options.override(.{ .color_text = opts.color(.press_text) });
                                                                                  ^
/home/sugar/SourceCode/keypass/src/gui/login.zig:103:28: 0x36da83 in login_frame (passkeez)
        if (try dvui.button(@src(), "Unlock", .{}, .{
                           ^
/home/sugar/SourceCode/keypass/src/gui.zig:79:34: 0x3964b2 in dvui_frame (passkeez)
        .login => try login_frame(),
                                 ^
/home/sugar/SourceCode/keypass/src/main.zig:88:27: 0x39d21b in main (passkeez)
        try gui.dvui_frame();
                          ^
/usr/local/bin/zig-linux-x86_64-0.11.0/lib/std/start.zig:574:37: 0x39f02e in main (passkeez)
            const result = root.main() catch |err| {
                                    ^
???:?:?: 0x7f3c120c8ccf in ??? (libc.so.6)
Unwind information for `libc.so.6:0x7f3c120c8ccf` was not available, trace may be incomplete

run passkeez: error: the following command terminated unexpectedly:
/home/sugar/SourceCode/keypass/zig-out/bin/passkeez 
Build Summary: 8/10 steps succeeded; 1 failed (disable with --summary none)
run transitive failure
└─ run passkeez failure
error: the following build command failed with exit code 1:
/home/sugar/SourceCode/keypass/zig-cache/o/1cab198d13bc6a121cff9c891e060197/build /usr/local/bin/zig-linux-x86_64-0.11.0/zig /home/sugar/SourceCode/keypass /home/sugar/SourceCode/keypass/zig-cache /home/sugar/.cache/zig run
```

Usually I compile with `-Doptimize=ReleaseSmall` so this is the first time I encountered the error but it's still present in the most recent commit (1220e30b896597e843a23df3c15e9b126ed4a3d0fb042bfb7c85f7c10827bd6e66a7).

**Comment by r4gus** [comment]
So this happens if you don't set the `press_text` field for one of the style structs.

```zig
.style_accent = .{
        .accent = accent.lighten(0.3),
        .fill = accent,
        .text = Color.white,
        .border = Color.lerp(accent, 0.4, Color.white),
        .hover = Color.lerp(accent, 0.2, Color.white),
        .press = Color.lerp(accent, 0.3, Color.white),
        .press_text = Color.white, // <= IMPORTANT
    },
```

I think, defaulting the fields to `null` (e.g. `press_text: ?Color = null,`) and `orelse unreachable` doesn't work well together.

**Comment by david-vanderson** [comment]
@r4gus That is definitely a bug and a footgun, I'm sorry.  I've been meaning to make the theme/colors stuff better, and this is a good opportunity to improve things.  This is my top priority right now.

I believe you are the trailblazer in terms of using your own theme.  Can you answer these:
* Do you ever use the color_style `.accent`, `.success`, or `.err`?  I'm thinking the current color_style thing isn't the right abstraction.
* I'm planning on making just a few colors mandatory, and the others optional (if not given we calculate something from the mandatory colors).  When making your theme, which colors did you need to change?

Thanks, and sorry again.

**Comment by r4gus** [comment]
To answer you question, I don't use the different color styles that much. I copied the default theme a few weeks back and made some adjustments to it. Now I'm manly playing with the overall layout of my application. And again, you really don't have to apologize for that.

**Comment by david-vanderson** [comment]
Thanks for the info - looks like color_style is going to go.  Oh I also wondered about your build.zig.zon file specifying the branch.  Is that so you don't have to manually update the revision all the time?  But you still have to update the hash right?

**Comment by david-vanderson** [comment]
@iacore Thanks for the quick improvement in the error messaging!

**Comment by r4gus** [comment]
> Thanks for the info - looks like color_style is going to go. Oh I also wondered about your build.zig.zon file specifying the branch. Is that so you don't have to manually update the revision all the time? But you still have to update the hash right?

Yes, by tracking master I just have to update the hash each time I make a change but I also use ezpkg from time to time.

**Comment by david-vanderson** [comment]
I've committed a switch to a simpler approach to colors.  Got rid of the `color_style` indirection.  Now when you want to specify colors in `Options`, you have 3 things:
* directly specify color
  * `.color_text = .{ .color = .{ .r = 0x35, .g = 0x84, .b = 0xe4 }`
* specify by name (to pick from theme)
  * `.color_fill = .{ .name = .fill_window }`
* use theme style
  * `_ = try dvui.button(@src(), "Accent", .{}, dvui.themeGet().style_accent);`

What do you think?  I'm slightly annoyed by the extra parentheses needed due to the union, but it seems much simpler at least to me.

`Theme` is now much simpler as well, having just a flat list of required `Color`s plus a few `Option`s.  @r4gus  I'd appreciate any feedback about how tough it is to modify your theme for this.

If this seems good, then I think I'll do the same for the font_style.

**Comment by r4gus** [comment]
Hey, sure. I'll give you feedback as soon as I've updated my theme.

**Comment by iacore** [comment]
> Yes, by tracking master I just have to update the hash each time I make a change but I also use ezpkg from time to time. I'll change that with the first main release.

I don't know about ezpkg before. I just do `ln -s ~/path/to/repo ~/.cache/zig/p/HASH` and git uses that.
